xrpl/wallet/wallet_generation.py

```diff
@@ -48,33 +59,41 @@ def generate_faucet_wallet(client: Client, debug: bool = False) -> Wallet:
         starting_balance = 0
 
     # Ask the faucet to send funds to the given address
-    post(url=FAUCET_URL, json={"destination": address})
+    post(url=faucet_url, json={"destination": address})
     # Wait for the faucet to fund our account or until timeout
     # Waits one second checks if balance has changed
-    # If balance doesn't change it will attempt again until timeout_seconds
-    for _ in range(timeout_seconds):
+    # If balance doesn't change it will attempt again until _TIMEOUT_SECONDS
+    i = 0
+    while i < _TIMEOUT_SECONDS:
         sleep(1)
         try:
             current_balance = get_balance(address, client)
         except XRPLRequestFailureException:
             current_balance = 0
-        # If our current balance has changed, then return
-        if starting_balance != current_balance:
+        # If our current balance has changed, then finish
+        if current_balance > starting_balance:
             if debug:
                 print("Faucet fund successful.")
-            try:
-                wallet.next_sequence_num = get_next_valid_seq_number(address, client)
-            except XRPLRequestFailureException as e:
-                if e.error_code != "actNotFound":
-                    raise
-                # try again after waiting a bit
-                sleep(1)
-                wallet.next_sequence_num = get_next_valid_seq_number(address, client)
+            break
+        i += 1
```

Comment:
not understanding why it was necessary to move to a `while` loop here?

Reply:
oh i see it's to have a global timeout. this isn't a big deal, but what do you think about this refactor:

```python
is_funded = false
for _ in range(_TIMEOUT_SECONDS):
  sleep(1)
  if not is_funded:
    is_funded = _try_to_fund()
  else:
    next_seq_num = _try_to_get_next_seq()
    if next_seq_num is not None:
      wallet.net_sequence_num = nex_seq_num
      return
```

Reply:
not a big deal - more of a suggestion. this way you can have:

1) only one loop relating to the same TIMEOUT, and not have to keep track of elapsed seconds
2) move the actual network calls to nice small functions (you know how much i like those lol)

Reply:
I agree with this suggestion. The starting balance isn't strictly necessary, but the sequence number is (as long as we get it from validated data, since if you rely on non-validated data then #3795 applies again)